.github/workflows/gradle.yml

```diff
@@ -1,68 +1,262 @@
-# This workflow will build a Java project with Gradle
-# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
-
+#
+# Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
 name: build
 
+# Set up when this workflow will run.
 on:
   push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
+    branches:
+      - master                 # On every push to the master branch.
+      - 'r[0-9]+.[0-9]+'       # On every push to a release branch.
```

Comment:
We don't get the full benefit of this regex, as can be seen during the Artifactory Snapshot step below: https://github.com/pravega/pravega/pull/5563/files#diff-0d634959c8276ae976e39ba475e63d0b9ec27824470c79f1971f58fe37c46325R241

NOTE: The gate for the Artifactory Snapshot step will need to be revisited when Pravega reached version 2.0.

Reply:
This particular wildcard will work for any future release. I think you are referring to the `snapshot` job (at the bottom) where I cannot use wildcard matches. There is no support for that in the `job.if` syntax. Perhaps by the time we reach version `2.x` (if ever - depending on our versioning scheme), GHA will add support for that.